src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,133 @@
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor PreferStreamReadAsyncMemoryOverloadsRule = DiagnosticDescriptorHelper.Create(
+                                                                                        RuleId,
+                                                                                        s_localizableTitleRead,
+                                                                                        s_localizableMessageRead,
+                                                                                        DiagnosticCategory.Performance,
+                                                                                        RuleLevel.IdeSuggestion,
+                                                                                        s_localizableDescriptionRead,
+                                                                                        isPortedFxCopRule: false,
+                                                                                        isDataflowRule: false);
+
+        internal static DiagnosticDescriptor PreferStreamWriteAsyncMemoryOverloadsRule = DiagnosticDescriptorHelper.Create(
+                                                                                        RuleId,
+                                                                                        s_localizableTitleWrite,
+                                                                                        s_localizableMessageWrite,
+                                                                                        DiagnosticCategory.Performance,
+                                                                                        RuleLevel.IdeSuggestion,
+                                                                                        s_localizableDescriptionWrite,
+                                                                                        isPortedFxCopRule: false,
+                                                                                        isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(PreferStreamReadAsyncMemoryOverloadsRule, PreferStreamWriteAsyncMemoryOverloadsRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            if (context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemory1) == null ||
```

Comment:
I think that's better to use the `TryGetOrCreateTypeByMetadataName`, `!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemory1, out _)`

Reply:
I decided to not use the `Try` method because I am not going to consume the `out` argument. Also, the method won't throw an exception on failure, so it's less code.

Reply:
I'll switch to Try, it looks nicer when they are aligned 😃